modules/luci-base/luasrc/dispatcher.lua

```diff
@@ -406,7 +406,7 @@ function dispatch(request)
 				return
 			end
 
-			http.header("Set-Cookie", 'sysauth=%s; path=%s' %{ sid, build_url() })
+			http.header("Set-Cookie", 'sysauth=%s; path=%s; HttpOnly; secure' %{ sid, build_url() })
```

Comment:
The `secure` attribute should only be enabled if `HTTPS` is on, otherwise browsers will ignore `Set-Cookie` and users cannot login.  It is indeed the case when I patched 17.01.04 armvirt qemu image to test this.